static/js/controllers/reports.js

```diff
@@ -480,6 +480,8 @@ var _ = require('underscore'),
           var reportId = target.attr('data-record-id');
           var checkbox = target.find('input[type="checkbox"]');
           var alreadySelected = _.findWhere($scope.selected, { _id: reportId });
+          // timeout so if the user clicked the checkbox it has time to
+          // register before we set it to the correct value.
```

Comment:
When a user clicks on a checkbox the browser toggles the checkbox state. If I go ahead and set checked to true then the browser will toggle it immediately afterwards. Instead I wait for the browser to do its thing then set it to what I want. The alternative is to register a second event listener on the checkbox itself which also does this code but also prevents default. I opted for the option which means fewer event listeners.